mitodl/open-discussions
# Assign url, and save full url path as run_slug, for ocw courses

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #3575 

#### What's this PR do?
Saves full url path for OCW courses (related to https://github.com/mitodl/ocw-studio/issues/1336) and assigns url field too.

#### How should this be manually tested?
- Set `OCW_NEXT_BASE_URL=https://ocwnext-rc.odl.mit.edu/`
- Run `manage.py backpopulate_ocw_next_data --overwrite --course-url-substring 15-912-technology-strategy-fall-2008`
- Check that the `run_slug` for the associated `LearningResourceRun` object is `courses/15-912-technology-strategy-fall-2008` and the `url` value is `https://ocwnext-rc.odl.mit.edu/courses/15-912-technology-strategy-fall-2008`
- Verify that your new course shows up in search results, and when you click on it for more details, a "Take this course" link is visible and leads to the correct url.
- Create and publish a new course in ocw-studio-rc.  Repeat the above using the appropriate `--course-url-substring` parameter.

On RC and prod, `manage.py backpopulate_ocw_next_data --overwrite` should be run to reimport everything.
